src/file_system/mod.rs

```diff
@@ -0,0 +1,345 @@
+use nonempty::NonEmpty;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Label(pub String);
+
+impl Label {
+    pub fn root_label() -> Self {
+        "~".into()
+    }
+}
+
+impl From<&str> for Label {
+    fn from(item: &str) -> Self {
+        Label(item.into())
+    }
+}
+
+impl From<String> for Label {
+    fn from(item: String) -> Self {
+        Label(item)
+    }
+}
+
+pub struct Path(pub NonEmpty<Label>);
```

Comment:
I guess you've looked at `std::path::PathBuf` already?

Reply:
I have not :eyes: I'll stick it on my todo list to check it out. Might be useful as a back-end for this implementation :thinking:

Reply:
One thing that we want to keep as an invariant is that `Directory` should not be empty, since an empty directory does not exist in VCSes

Reply:
But this is a path, not a directory? Or am I misunderstanding?

Reply:
Yeah true, the `Directory` is already non-empty :thinking:

Reply:
`Path` is the way to specify where the user will be searching in a `Directory` though

Reply:
So if we're searching a directory there's no point in supplying an empty path

Reply:
Created an issue for this https://github.com/radicle-dev/radicle-surf/issues/9 so we don't forget